### DISCUSSION TITLE
How to pass variables into `VERSIONS`

### DISCUSSION THREAD

**Discussion by Mark2Mark** [coldtype/coldtype#164] [Q&A]
```py
font_a = "Foo.ttf"
font_b = "Bar.ttf"

VERSIONS = {
    "Foo": dict(font=font_a),
    "Bar": dict(font=font_b),
} #/VERSIONS
```

This king of variable-passing does not work. Since I don’t understand the implementation of this `VERSIONS` dictionary in CT, I cannot wrap my head around it without asking you for more insights on that, please. Thanks so much in advance :)

**Comment by stenson** [comment]
Totally correct, variables will not work there. The inline `VERSIONS` are meant for literal-only code that can be eval'd without reference to any other code (since the `VERSIONS = ... #/VERSIONS` bit is literally excerpted from the code and eval'd directly).

The solution to this is to define a second file with the same name as your coldtype source file, but with _versions.py as the ending. That way you can define a VERSIONS dictionary completely dynamically.

Here’s a pair of files from the examples/animations folder in the coldtype repo:

<img width="1288" alt="image" src="https://github.com/coldtype/coldtype/assets/89770/68c9b564-595a-4b2f-9c32-4d6b2e1d9c73">

All that said, `:=` variables can actually work in inline VERSIONS if they're specified completely inline without reference to other code, so something like this should work:

```python
VERSIONS = {
    "A": dict(text="COLD", font=(font:="Times")),
    "B": dict(text="TYPE", font=font),
} #/VERSIONS
```

Not sure that's particularly useful, but maybe worth pointing out.

**Comment by Mark2Mark** [comment]
Thanks Rob!

I’ll see if I can manage the sidecar thingy.

BTW, what was that syntax with the double degree signs wrapping something? (the image seems gone now, but I saw that before).

**Reply by stenson** [reply]
oh the `ºº` is just an experiment in getting VS Code to not mark a variable as undefined when it's defined dynamically by coldtype, nothing importan! exactly the same as `__` for anything coldtype-related

**Reply by Mark2Mark** [reply]
I see! Oh yeah, I deteste that undefined linter errors. However, the `__` dunder is not necessary coldtype-related. That’s usually python stuff. That’s why I was also confused about the implementation of the `__VERSION__` thingy :)
